File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        
        countS = {}
        countT = {}
        
        for i in range(len(s)):
            charS = s[i]
            charT = t[i]
            countS[charS] = countS.get(s[i], 0) + 1
            countT[charT] = countT.get(t[i], 0) + 1
        return countS == countT

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    let hash = {}
    if (s.length !== t.length) return false

    for(let i = 0; i < s.length; i++) {
        hash[s[i]] = (hash[s[i]] || 0) + 1;
    }
    for(let i = 0; i < t.length; i++) {
        if (!hash[t[i]]) return false;
        hash[t[i]]--;
        // console.log(hash)
    }
    return true
    
};